src/time/TimeGrids.cpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <numeric>
+#include <vector>
+
+#include "TimeGrids.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice::time {
+
+void TimeGrids::saveTimeGridOfCouplingData(const DataMap &cplData, std::vector<int> dataIDs, bool reducedTimeGrid)
```

Comment:
The key of the cplData includes both primary and secondary data. I added the dataIDs parameter to be able to only save the time grids in primary data in _primaryTimeGrids instance, but I can remove this if it simplifes the code.

Reply:
I have the saveTimeGridOfCouplingData to become the constructor of this class. To do this I also changed the _timeGrids and _primaryTimeGrids to unique pointers. It would be nice to get some feedback on this, since I am not that comfortable with unique pointers.